app/components/post_list/post_list.ios.js

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FlatList, StyleSheet} from 'react-native';
+
+import {ListTypes} from 'app/constants';
+import {makeExtraData} from 'app/utils/list_view';
+
+import PostListBase from './post_list_base';
+
+const INITIAL_BATCH_TO_RENDER = 15;
+const SCROLL_UP_MULTIPLIER = 3.5;
+
+export default class PostList extends PostListBase {
+    constructor(props) {
+        super(props);
+
+        this.hasDoneInitialScroll = false;
+        this.contentOffsetY = 0;
+        this.makeExtraData = makeExtraData();
+
+        this.state = {
+            managedConfig: {},
+            postListHeight: 0,
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.channelId !== nextProps.channelId) {
+            this.contentOffsetY = 0;
+            this.hasDoneInitialScroll = false;
+        }
+    }
+
+    handleContentSizeChange = (contentWidth, contentHeight) => {
+        this.scrollToInitialIndexIfNeeded(contentWidth, contentHeight);
+
+        if (this.state.postListHeight && contentHeight < this.state.postListHeight && this.props.extraData) {
+            // We still have less than 1 screen of posts loaded with more to get, so load more
+            this.props.onLoadMoreUp();
+        }
+    };
+
+    handleLayout = (event) => {
+        const {height} = event.nativeEvent.layout;
+        if (this.mounted) {
```

Comment:
Can this be called after the component is unmounted? This check seems unnecessary